olwebsite-app/src/pages/RegisterView.jsx

```diff
@@ -1,30 +1,137 @@
-import React from "react";
-import { Link, withRouter } from "react-router-dom";
+import React, { useState } from "react";
+import { Link, Redirect, withRouter } from "react-router-dom";
+
+import { 
+  validateEmail, 
+  validatePassword, 
+  validateReenteredPassword 
+} from '../utils/authentication-utils';
+import { SUCCESS } from '../constants/authentication-constants';
+import { REGISTER_PATH } from '../constants/rest-constants';
 
 import Input from '../components/Objects/Input/Input';
 import homeLogo from "../images/homeLogo.png";
 
 function RegisterView() {
+  /************************************
+   * State
+   ************************************/
+
+  const [isRegistered, setIsRegistered] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const [email, setEmail] = useState('');
+  const [emailErrorMessage, setEmailErrorMessage] = useState(null);
+  const [password, setPassword] = useState('');
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState(null);
+  const [reenteredPassword, setReenteredPassword] = useState('');
+  const [reenteredPasswordErrorMessage, setReenteredPasswordErrorMessage] = useState(null);
+
+  /************************************
+   * Private Functions
+   ************************************/
+
+ /* Handler for sign up button clicked, if valid inputs registers user
+  */
+  async function registerClicked() {
+    if (validateInput()) {
+      await registerUser();
```

Comment:
I think a spinner/loading screen would be nice especially if the call is hanging for whatever reason. Do you want me to open a new issue?

Reply:
Sounds good to me